sui-execution/latest/sui-adapter/src/programmable_transactions/execution.rs

```diff
@@ -829,14 +846,24 @@ mod checked {
             }
         }
         // script visibility checked manually for entry points
-        let mut result = context
-            .execute_function_bypass_visibility(
+        let mut result = if trace_builder_opt.is_some() {
+            context.execute_function_bypass_visibility(
```

Comment:
I think since it's using the same entrypoint this `if` is no longer needed here

Reply:
Good point! I was looking at it without actually seeing it...